### PR TITLE
Add pagination support to all functions

### PR DESCRIPTION
**Pull request by aurelienhess**
- Replaces custom_endpoint() by sleeker _get_request() function
- Fixed the mess with the get_rules() function. However, for backwards compatibility's sake, some weirdness had to remain
- This commit includes the groups endpoints which are part of another open pull request
- get_rules() without parameters now returns an error. As the use of this function was only used internally by get_all_rules() we can discard it IMO, as get_all_rules now now longer depends on this.

### CODE REVIEW

File: modules/vectra.py [hunk 1]
```diff
@@ -522,6 +569,122 @@ def delete_rule(self, rule_id=None, restore_detections=True):
...
+        return requests.get('{url}/groups/{id}'.format(url=self.url, id=group_id), headers=self.headers, verify=False)
+
+    @validate_api_v2
+    def get_all_groups(self):
+        """
+        Get all existing get_all_groups
+        Groups do not implement pagination
+        For consistency's sake, we still implement a generator
+        """
+        yield requests.get('{url}/groups'.format(url=self.url), headers=self.headers, verify=False)
```
Comment (HoneyBaked) [inline modules/vectra.py:588]: This endpoint should have pagination added, similar to the other endpoints, if it does, then we need to use 'next..

Comment (aurelienhess) [reply modules/vectra.py:588]: But this is not the case now, right? I'd say we add pagination support on the API tools when the endpoint exposes it, no?

Comment (HoneyBaked) [reply modules/vectra.py:588]: Yeah we should update it in this PR get the groups from resp.json()['results'] and iterate through the pages similar to get_hosts.

**Comment by aurelienhess** [comment]
Splitting into separate pull requests

**Pull request closed without merging**